packages/prettier-plugin-java/src/cst-printer.js

```diff
@@ -79,7 +79,17 @@ class CstPrettierPrinter extends BaseJavaCstVisitor {
 
       if (ctx.ignore) {
         try {
-          return buildOriginalText(retrieveNodesToken(ctx), this.originalText);
+          return buildOriginalText(
+            getCSTNodeToken(
```

Comment:
if this is an "heavy" process why do we call it twice?
clone and sort externally from the "heavy" collection 😄

Reply:
Ah I see it returns only a single token, but iterates over the structure twice.
I would try to make it return two tokens (first and last) in one call and avoid the multiple iterations.

Reply:
I was trying to make a "generic" function bu it seems that we should not in this case.
Corrected as you suggested.